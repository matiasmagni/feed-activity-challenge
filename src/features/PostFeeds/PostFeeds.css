i.fas.fa-comments {
    color: #ffffff;
}
