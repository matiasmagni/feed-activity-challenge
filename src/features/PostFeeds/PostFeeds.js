import React, { useState, useEffect } from 'react';
import { Link } from "react-router-dom";
import axios from 'axios';
import config from '../../config';
import './PostFeeds.css';

const PostFeeds = () => {
    const [posts, setPosts] = useState([]);

    useEffect(() => {
        const fetchPosts = async () => {
            const result = await axios(config.API_URL.POSTS);
            setPosts(result.data);
        };

        fetchPosts();
    }, []);

    return (
        <div className="container-fluid video-player">
            <div className="container">
                <div className="row vr-gallery">
                    {posts.map(post => (
                        <div className="row" key={`post-${post.id}`}>
                            <div className="col-md-8 mb-4">
                                <div className="row">
                                    <div className="col-md-12 col-lg-7 pr-0 pd-md">
                                        <img src="images/gallery-img1.png" alt="" />
                                    </div>
                                    <div className="col-md-12 col-lg-5 light-bg cus-pd cus-arrow-left">
                                        <p><small>march 27, 2020</small></p>
                                        <h3>{post.title}</h3>
                                        <p>{post.body}</p>
                                    </div>
                                </div>
                            </div>
                            <div className="col-md-4 pl-4 mb-4">
                                <div className="card">
                                    <img className="card-img h-100" src="images/video-cover1.jpg" alt="" />
                                    <div className="card-img-overlay opacity text-center">
                                        <Link className="play-1" to={`/comments/${post.id}`}>
                                            <i className="fas fa-comments fa-2x"></i>
                                        </Link>
                                        <h5 className="card-title">View all comments...</h5>
                                    </div>
                                </div>
                            </div>
                        </div>
                    ))}
                    <div className="col-md-12 text-center">
                        <a href="/#" className="btn">LOAD MORE</a>
                    </div>
                </div>
            </div>
        </div>
    );
}

export default PostFeeds;