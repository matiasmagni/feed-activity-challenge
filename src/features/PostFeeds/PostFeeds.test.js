import React from 'react';
import { act as domAct } from "react-dom/test-utils";
import { act as testAct, create } from "react-test-renderer";
import PostFeeds from './PostFeeds';

describe('PostFeeds component tests.', () => {
  test('PostFeeds component renders correctly.', () => {
    let tree;
  
    domAct(() => {
      testAct(() => {
        tree = create(<PostFeeds />);
      });
  
      expect(tree).toMatchSnapshot();
    });
  });
});
