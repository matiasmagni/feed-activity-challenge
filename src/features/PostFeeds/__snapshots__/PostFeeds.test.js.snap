// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostFeeds component tests. PostFeeds component renders correctly. 1`] = `
<div
  className="container-fluid video-player"
>
  <div
    className="container"
  >
    <div
      className="row vr-gallery"
    >
      <div
        className="col-md-12 text-center"
      >
        <a
          className="btn"
          href="/#"
        >
          LOAD MORE
        </a>
      </div>
    </div>
  </div>
</div>
`;
